pkg/kubelet/kubelet.go

```diff
@@ -170,16 +170,30 @@ func (kl *Kubelet) GetRootDir() string {
 
 // GetPodsDir returns the full path to the directory under which pod
 // directories are created.
-// TODO(thockin): For now, this is the same as the root because that is assumed
-// in other code.  Will fix.
 func (kl *Kubelet) GetPodsDir() string {
-	return kl.GetRootDir()
+	return path.Join(kl.GetRootDir(), "pods")
 }
 
 // GetPodDir returns the full path to the per-pod data directory for the
 // specified pod.  This directory may not exist if the pod does not exist.
 func (kl *Kubelet) GetPodDir(podUID string) string {
```

Comment:
I'm not sure I follow.  GetRootDir() is controlled by kubelet, so any trickery would have to come through the flag that controls root dir.  If you are setting flags you are already root.  The joining of pod UID and container name predicate on the validation done before we accept a pod.

Can you clarify what sort of attack you are conceiving of?  Caveat, I suck at security analysis.